lib/webapp.js

```diff
@@ -1005,6 +1020,15 @@ WebServiceHandle.prototype = {
       }
       //if not internal routing
       if (httpOrHttps) {
+        // Removing the version from requestOptions.path if request is for zss
+        if(requestOptions.hostname === this.environment.agentRequestOptions.host && requestOptions.port === this.environment.agentRequestOptions.port) {
+          const versionPattern = new RegExp('^(\\d+\\.)?(\\d+\\.)?(\\*|\\d+)$');
+          const reqPath = requestOptions.path;
+          const reqPathVersion = reqPath.substring(reqPath.lastIndexOf('/') + 1);
```

Comment:
Here's a few types of URLs that can exist in this server
/plugins/my.plugin.name/services/abc/1.2.3/
/plugins/my.plugin.name/services/abc/1.2.3/some-rest-api/4.5.6
/plugins/my.plugin.name/services/abc/_current/
/plugins/my.plugin.name/services/abc/_current/some-rest-api/4.5.6

I believe you are only considering case 1 here, and accidentally also doing something on cases 2 and 4.
What you should instead do is check if there's a semver or "_current" in position 5 of the url.
Also, the regex pattern for semver is defined on semver.org as 
`^(0|[1-9]\d*)\.(0|[1-9]\d*)\.(0|[1-9]\d*)(?:-((?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*)(?:\.(?:0|[1-9]\d*|\d*[a-zA-Z-][0-9a-zA-Z-]*))*))?(?:\+([0-9a-zA-Z-]+(?:\.[0-9a-zA-Z-]+)*))?$`
truly a large regex, but the correct one.

Reply:
This PR is outdated and needs to be closed.
The issues in this PR are already addressed and merged. Also the regex for the semverion that we are currently using is correct but to be more precise I will update the regex with the one you have mentioned above.